src/data/nat/squarefree.lean

```diff
@@ -513,7 +513,8 @@ end
 
 /-- Evaluates the `squarefree` predicate on naturals. -/
 @[norm_num] meta def eval_squarefree : expr → tactic (expr × expr)
-| `(@squarefree ℕ _ %%e) := do
+| `(@squarefree ℕ %%inst %%e) := do
```

Comment:
I would speculate you can actually get rid of `ℕ` here, since all that matters is that the instance matches.

The test would be whether you can solve `@squarefree (multiplicative ℕ) nat.monoid (3 : ℕ)`.

Of course it would be totally reasonable to just refuse to solve this.

Reply:
I think failing the `is_def_eq` check could potentially take a long time for complicated types so I'd rather fail quickly and exclude some cases that could have worked but should not really appear anyway.

Reply:
I can only assume something has gone wrong with github:

![image](https://user-images.githubusercontent.com/425260/229279698-71df0792-f5a9-4ad0-af3a-e3f8e1b9518d.png)

unless you've merged your identity with the [Square](https://github.com/Square) payment provider!

Reply:
Interesting! Maybe it's some sort of automatic translation going on (vierkant being Dutch for square)? Or just a weird april fools thing?

Reply:
Ah you're right, this was my browser trying to translate for me!